### ISSUE TITLE
False Negative: Non-canonical forms of empty `finalize` methods are not matched.

### ISSUE THREAD

**Issue by Carlson-JLQ** [pmd/pmd#6529]
# False Negative: Non-canonical forms of empty `finalize` methods are not matched.

Version
7.22.0

## Checker
- Checker id: `category/java/errorprone.xml/EmptyFinalizer`
- Checker description: Detects empty finalize methods in non-final classes or final classes with non-final finalize methods.

## Description of the false negative
`diff/output` marks 1 false negatives for `category/java/errorprone.xml/EmptyFinalizer`. Across these samples, the benchmark is still exercising the same underlying bug pattern, but the implementation appears to lose track of it once the source deviates from one canonical shape. The recurring themes are suppression or annotations.

### `PosCase2.java`
Source: `output/codeql/EmptyFinalize/src/main/java/scensct/core/pos/PosCase2.java`

The sample keeps the benchmark's target pattern but expresses it in a slightly non-canonical source form. The risky pattern is still present in the code, so the absence of a report points to a matching gap rather than a benign variant. That is also how the benchmark classifies the sample: A class defines a finalize method whose body consists solely of an empty block statement should be flagged as empty finalize.

```java
// A class defines a finalize method whose body consists solely of an empty block statement should be flagged as empty finalize.
package scensct.core.pos;

public class PosCase2 {
    // Finalize method with empty block statement
    @Override
    protected void finalize() throws Throwable { // [REPORTED LINE]
        {} // Solely an empty block statement
    }
}
```

## Cause analysis
The rule is documented as follows: Detects empty finalize methods in non-final classes or final classes with non-final finalize methods.

In this set the problem shows up around suppression or annotations. The missed cases suggest that the implementation is narrower than the rule description and too dependent on one preferred AST shape.

Once the same behavior is expressed through a helper, an overload, a modifier such as `abstract` or `native`, a different comment position, or a slightly rearranged control-flow structure, the report disappears even though the benchmark still treats the case as in-scope.

For `category/java/errorprone.xml/EmptyFinalizer`, the practical improvement would be to generalize the match so it follows the underlying behavior rather than one exact spelling of it.

## References
None known.

**Comment by oowekyala** [comment]
An empty block is its own issue (EmptyControlStatement).
